### PR TITLE
create a script autocompletion

### PR DESCRIPTION
**Pull request by nashousni**

### CODE REVIEW

File: gse-afs-ext-base/src/main/java/com/powsybl/gse/afs/ext/base/ActionScriptKeywordsExtension.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+public class ActionScriptKeywordsExtension implements AutoCompletionWordsProvider {
+
+    @Override
+    public Class getProjectFileType() {
+        return ActionScript.class;
+    }
+
+    @Override
+    public Set<String> completionKeyWords() {
+        return new HashSet<>(Arrays.asList("action", "apply", "contingency", "description", "equipments", "life", "rule", "tasks", "test", "when"));
```
Comment (geofjamg) [inline gse-afs-ext-base/src/main/java/com/powsybl/gse/afs/ext/base/ActionScriptKeywordsExtension.java]: Could be a constant?

File: gse-util/src/main/java/com/powsybl/gse/util/AutoCompletion.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            if (event.getClickCount() == 2) {
+                menuItemOnAction(tokenLength);
+            }
+        });
+
+        /* * Each row in a ListView should be 24px tall.  Also, we have to add an extra
+         * two px to account for the borders of the ListView.
+         * ScrollPane is set to be visible when Items size exceed MAX_ITEMS_SIZE
+         */
+        final int rowHeight = 24;
```
Comment (geofjamg) [inline gse-util/src/main/java/com/powsybl/gse/util/AutoCompletion.java]: Could be a constant?

File: gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java [hunk 3]
```diff
@@ -67,10 +80,24 @@
 
     private static final ServiceLoaderCache<KeywordsProvider> KEYWORDS_LOADER = new ServiceLoaderCache<>(KeywordsProvider.class);
 
+    private static final ServiceLoaderCache<AutoCompletionWordsProvider> AUTO_COMPLETION_WORDS_LOADER = new ServiceLoaderCache<>(AutoCompletionWordsProvider.class);
```
Comment (geofjamg) [inline gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java]: Why is it both cached in ModificationScriptEditor and GroovyCodeEditor?

File: gse-afs-ext-base/src/main/java/com/powsybl/gse/afs/ext/base/ActionScriptKeywordsExtension.java (hunk 1 above)
Comment (nashousni) [reply gse-afs-ext-base/src/main/java/com/powsybl/gse/afs/ext/base/ActionScriptKeywordsExtension.java]: updated

File: gse-util/src/main/java/com/powsybl/gse/util/AutoCompletion.java (hunk 2 above)
Comment (nashousni) [reply gse-util/src/main/java/com/powsybl/gse/util/AutoCompletion.java]: updated

File: gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java (hunk 3 above)
Comment (nashousni) [reply gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java]: updated

**Pull request merged**